spec.html

```diff
@@ -32043,92 +32043,269 @@ <h1>Time Values and Time Range</h1>
-        </emu-eqn>
-      </emu-clause>
-
-      <emu-clause id="sec-week-day">
-        <h1>Week Day</h1>
-        <p>The weekday for a particular time value _t_ is defined as</p>
-        <emu-eqn aoid="WeekDay">WeekDay(_t_) = 𝔽(ℝ(Day(_t_) + *4*<sub>𝔽</sub>) modulo 7)</emu-eqn>
-        <p>A weekday value of *+0*<sub>𝔽</sub> specifies Sunday; *1*<sub>𝔽</sub> specifies Monday; *2*<sub>𝔽</sub> specifies Tuesday; *3*<sub>𝔽</sub> specifies Wednesday; *4*<sub>𝔽</sub> specifies Thursday; *5*<sub>𝔽</sub> specifies Friday; and *6*<sub>𝔽</sub> specifies Saturday. Note that <emu-eqn>WeekDay(*+0*<sub>𝔽</sub>) = *4*<sub>𝔽</sub></emu-eqn>, corresponding to Thursday, 1 January 1970.</p>
+      <emu-clause id="sec-time-related-constants">
+        <h1>Time-related Constants</h1>
+        <p>These constants are referenced by algorithms in the following sections.</p>
+        <emu-eqn id="eqn-HoursPerDay" aoid="HoursPerDay">HoursPerDay = 24</emu-eqn>
+        <emu-eqn id="eqn-MinutesPerHour" aoid="MinutesPerHour">MinutesPerHour = 60</emu-eqn>
+        <emu-eqn id="eqn-SecondsPerMinute" aoid="SecondsPerMinute">SecondsPerMinute = 60</emu-eqn>
+        <emu-eqn id="eqn-msPerSecond" aoid="msPerSecond">msPerSecond = *1000*<sub>𝔽</sub></emu-eqn>
+        <emu-eqn id="eqn-msPerMinute" aoid="msPerMinute">msPerMinute = *60000*<sub>𝔽</sub> = msPerSecond × 𝔽(SecondsPerMinute)</emu-eqn>
+        <emu-eqn id="eqn-msPerHour" aoid="msPerHour">msPerHour = *3600000*<sub>𝔽</sub> = msPerMinute × 𝔽(MinutesPerHour)</emu-eqn>
+        <emu-eqn id="eqn-msPerDay" aoid="msPerDay">msPerDay = *86400000*<sub>𝔽</sub> = msPerHour × 𝔽(HoursPerDay)</emu-eqn>
+      </emu-clause>
+
+      <emu-clause id="sec-day" type="abstract operation" oldids="eqn-Day,sec-day-number-and-time-within-day">
+        <h1>
+          Day (
+            _t_: a finite time value,
+          ): an integral Number
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It returns the day number of the day in which _t_ falls.</dd>
+        </dl>
+        <emu-alg>
+          1. Return 𝔽(floor(ℝ(_t_ / msPerDay))).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-timewithinday" type="abstract operation" oldids="eqn-TimeWithinDay">
+        <h1>
+          TimeWithinDay (
+            _t_: a finite time value,
+          ): an integral Number in the interval from *+0*<sub>𝔽</sub> (inclusive) to msPerDay (exclusive)
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It returns the number of milliseconds since the start of the day in which _t_ falls.</dd>
+        </dl>
+        <emu-alg>
+          1. Return 𝔽(ℝ(_t_) modulo ℝ(msPerDay)).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-daysinyear" type="abstract operation" oldids="eqn-DaysInYear,sec-year-number">
+        <h1>
+          DaysInYear (
+            _y_: an integral Number,
+          ): *365*<sub>𝔽</sub> or *366*<sub>𝔽</sub>
```

Comment:
Let's return integers here instead of Numbers.

Reply:
They're Numbers in the status quo, so I'm inclined to leave them that way. Might as well leave type-changing to the follow-up PR you mentioned.

Reply:
Since we're using these as enums (only switching on them, never doing arithmetic on them or exposing them), I wouldn't include it in the arithmetic follow-up I mentioned. It's fine to not do it in this PR, but then I guess we should just have it be its own PR.

Reply:
I'd prefer to just refactor `DaysInYear` out of existence. (Not in this PR.)

Reply:
That also works for me.